solrmonitor/SolrEventListener.go

```diff
@@ -0,0 +1,18 @@
+package solrmonitor
+
+/**
+The client can register the SolrEventListener to listen to the solr cluster events from the zookeeper.
+ */
+type SolrEventListener interface {
+	// all the live nodes in the solr cluster
+	SolrLiveNodesChanged(livenodes []string)
+
+	// all the query aggregator nodes in ths solr cluster
+	SolrQueryNodesChanged(querynodes []string)
+
+	// all the collections in the solr cluster
+	SolrCollectionsChanged(collections []string)
+
+	// current collection state
+	SolrCollectionChanged(name string, collectionState *CollectionState)
```

Comment:
I would probably go one of two directions here:
a) open source our protos and double down on that
b) allow the caller to configure a deserializer to produce a specific concrete type, make the default the open source objects

Reply:
We can definitely take this task. As I mentioned, overhead of cloning from gosolr.collection object to our proto will be very less as we will have PRS callback and that will have update for fraction of replicas.